onnxruntime/core/optimizer/optimizer_execution_frame.h

```diff
@@ -49,6 +47,8 @@ class OptimizerExecutionFrame final : public IExecutionFrame {
     }
 
     const OpKernel* GetKernel(NodeIndex node_id) const;
+    Status RunSingleKernel(std::vector<int>& fetch_mlvalue_idxs, size_t node_index, std::vector<OrtValue>& fetches,
+                           const logging::Logger& logger = logging::LoggingManager::DefaultLogger());
 
```

Comment:
I don't think this belong in the Info class as that would give the class two different responsibilities (keeping the info for the frame and actually running a kernel), which is against SOLID principles. Probably better suited as a standalone utility method as it doesn't require any internals of OptimizerExecutionFrame does it?

Reply:
Where do you think is better?

Reply:
If this function is not used outside constant folding, why not just put it in constant_folding.cc as a static function? Or else some optimizer_utils?

Reply:
The whole file is just for constant folding. Should I delete this file too?